core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -419,6 +422,7 @@ private void unmanage(final Entity e, ManagementTransitionMode mode, boolean has
                 // migrating away or in-place active partial rebind:
                 ((EntityInternal)e).getManagementSupport().onManagementStopping(info);
                 stopTasks(e);
+                pruneOrphanedLocations(e);
```

Comment:
Not sure whether this is the right thing to do in this situation, as we don't actually remove the entity from the internal collections.